app/validators/types_de_champ/condition_validator.rb

```diff
@@ -1,13 +1,16 @@
 class TypesDeChamp::ConditionValidator < ActiveModel::EachValidator
   def validate_each(procedure, attribute, types_de_champ)
-    public_tdcs = types_de_champ.to_a
-      .flat_map { _1.repetition? ? procedure.draft_revision.children_of(_1) : _1 }
+    tdcs = if attribute == :draft_types_de_champ_private
+      procedure.draft_revision.types_de_champ_for
+    else
+      procedure.draft_revision.types_de_champ_for(scope: :public)
+    end
 
-    public_tdcs
+    tdcs
       .map.with_index
       .filter_map { |tdc, i| tdc.condition? ? [tdc, i] : nil }
       .map do |tdc, i|
-        [tdc, tdc.condition.errors(public_tdcs.take(i))]
+        [tdc, tdc.condition.errors(tdcs.take(i))]
       end
       .filter { |_tdc, errors| errors.present? }
       .each do |tdc, _error_hash|
```

Comment:
tiens à l'occas on pourra essayer de refactorer cet enchainement pour éviter autant de boucles et allocations

Reply:
peut-être que ça marche : 

```suggestion
    tdcs.each_with_index do |tdc, i|
      next unless tdc.condition?
    
      errors = tdc.condition.errors(public_tdcs.take(i))
      next if errors.blank?
    
      procedure.errors.add(
        attribute,
        procedure.errors.generate_message(attribute, :invalid_condition, { value: tdc.libelle }),
        type_de_champ: tdc
      )
    end
```